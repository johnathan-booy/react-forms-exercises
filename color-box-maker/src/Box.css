.Box-group {
	margin: 10px;
	display: flex;
	justify-content: center;
	align-items: center;
}

.Box {
	border: 1px solid black;
	border-radius: 10px;
	margin-right: 8px;
}

.Box-remove {
	background-color: white;
	border: 1px solid black;
	border-radius: 3px;
	cursor: pointer;
}

.Box-remove:hover {
	outline: 2px solid rgba(0, 0, 0, 0.213);
}
