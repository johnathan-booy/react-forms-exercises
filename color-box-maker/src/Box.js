import React from "react";
import "./Box.css";

const Box = ({ id, color, width, height, removeBox }) => {
	const style = {
		backgroundColor: color,
		width: `${width}px`,
		height: `${height}px`,
	};
	const handleRemove = (e) => {
		removeBox({ id, color, width, height });
	};
	return (
		<div className="Box-group" role="row">
			<div role="cell" className="Box" style={style}></div>
			<button className="Box-remove" onClick={handleRemove}>
				X
			</button>
		</div>
	);
};

export default Box;
