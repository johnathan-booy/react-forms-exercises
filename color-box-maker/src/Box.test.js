import { render } from "@testing-library/react";
import Box from "./Box";

const BOX = <Box id={1} color={"white"} width={100} height={100} key={1} />;

it("renders without crashing", () => {
	render(<Box />);
});

it("matches snapshot", () => {
	const { asFragment } = render(BOX);
	expect(asFragment()).toMatchSnapshot();
});

it("renders box with styles", () => {
	const { getByRole } = render(BOX);
	const box = getByRole("cell");
	expect(box).toHaveStyle("background-color:white; width:100px; height:100px");
});

it("renders delete button", () => {
	const { getByText } = render(BOX);
	const button = getByText("X");
	expect(button).toBeInTheDocument();
});
