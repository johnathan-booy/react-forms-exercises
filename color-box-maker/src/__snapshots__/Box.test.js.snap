// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box-group"
    role="row"
  >
    <div
      class="Box"
      role="cell"
      style="background-color: white; width: 100px; height: 100px;"
    />
    <button
      class="Box-remove"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
